akka-stream-tests/src/test/scala/akka/stream/io/TimeoutsSpec.scala

```diff
@@ -0,0 +1,238 @@
+package akka.stream.io
+
+import java.util.concurrent.TimeoutException
+
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl._
+import akka.stream.testkit.{ AkkaSpec, TestSubscriber, TestPublisher }
+import scala.concurrent.{ Future, Await }
+import scala.concurrent.duration._
+
+import akka.stream.testkit.Utils._
+
+class TimeoutsSpec extends AkkaSpec {
+  implicit val mat = ActorMaterializer()
+
+  "InitialTimeout" must {
+
+    "pass through elements unmodified" in assertAllStagesStopped {
+      Await.result(
+        Source(1 to 100).via(Timeouts.initalTimeout(2.seconds)).grouped(200).runWith(Sink.head),
+        3.seconds) should ===(1 to 100)
+    }
+
+    "pass through error unmodified" in assertAllStagesStopped {
+      a[TE] shouldBe thrownBy {
+        Await.result(
+          Source(1 to 100).concat(Source.failed(TE("test")))
+            .via(Timeouts.initalTimeout(2.seconds))
```

Comment:
I think I was wrong, this test is quick, it will normally complete with the TE failure immediately, right?